flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/controller/FlinkDeploymentController.java

```diff
@@ -109,6 +119,22 @@ public DeleteControl cleanup(FlinkDeployment flinkApp, Context context) {
         } catch (Exception e) {
             throw new ReconciliationException(e);
         }
+
+        if (!jobManagerDeployments.containsKey(flinkApp)) {
```

Comment:
Maybe we could extract the checking and loading of the deployment from the secondaryResource into a new method in the controller to keep the main reconcile flow as simple as possible

Reply:
We could also move the whole logic into JobStatusObserver passing the context object.

Reply:
It would scatter the resource watching logic. I would prefer to keep it here since it needs to work in tandem with `prepareEventSources`

Reply:
I also agree with @morhidi that this probably fits best into the observer itself because that is the component responsible for checking the running jobs already. The observer could also prepare the event sources that we would just pass in the controller.

Reply:
Refactored the logic and `JobStatusObserver` can be removed / consolidated with JobReconciler in a follow-up.